src/sdk/mod.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+// Copyright 2022-2023 Protocol Labs
+//! Ipc agent sdk, contains the json rpc client to interact with the IPC agent rpc server.
+
+mod subnet;
+mod wallet;
+
+use crate::jsonrpc::JsonRpcClientImpl;
+pub use wallet::LotusJsonKeyType;
+
+use url::Url;
+
+/// The ipc agent client
+pub struct IpcAgentClient<T> {
+    json_rpc_client: T,
+}
+
+impl<T> IpcAgentClient<T> {
```

Comment:
Should we bound a bit more the type so only Transports can be used here?

Reply:
I feel in rust the convention is the `impl` would be a the place to bound it, the struct definition is more general. But I could be wrong, but this is the pattern in many repos I saw in rust.